core-ui/src/components/Extensibility/helpers/index.js

```diff
@@ -89,6 +89,36 @@ export function createTemplate(api, namespace, scope) {
   return template;
 }
 
+export function getDefaultPreset(presets, emptyTemplate) {
+  if (!presets || !presets.length) return null;
+  const defaultPreset = presets.find(preset => preset.default === true);
+  return defaultPreset
+    ? merge({}, { value: emptyTemplate }, defaultPreset)
+    : null;
+}
+
+export function usePreparePresets(resource, presets) {
+  const { t: tExt } = useGetTranslation();
+  const { t } = useTranslation();
+
+  if (!presets || !presets.length) return null;
+  let preparedPresets = [];
+
+  presets.forEach(preset => {
+    preparedPresets.push(
+      merge({}, { value: resource }, { ...preset, name: tExt(preset.name) }),
+    );
+  });
```

Comment:
Why `forEach` instead of `map` function? With that approach, preparedPresets can be a `const`.